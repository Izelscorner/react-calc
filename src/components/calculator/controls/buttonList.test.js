import React from 'react';
import { mount } from 'enzyme';
import ButtonList from './buttonList';
import Button from './button';
import toJson from 'enzyme-to-json';

describe('<ButtonList />', () => {
  let wrapper;
  let handleEvent;

  beforeAll(() => {
    handleEvent = jest.fn();
    wrapper = mount(<ButtonList handleEvent={handleEvent}  />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the button and value', () => {
    const buttons = wrapper.find(Button);
    expect(buttons).toHaveLength(17);
  });

});
