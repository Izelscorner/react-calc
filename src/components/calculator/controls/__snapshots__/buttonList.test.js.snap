// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<ButtonList
  handleEvent={[MockFunction]}
>
  <div
    className="button-list"
  >
    <div
      className="button-list__row"
    >
      <Button
        handleEvent={[Function]}
        value="7"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          7
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="8"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          8
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="9"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          9
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="*"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          *
        </button>
      </Button>
    </div>
    <div
      className="button-list__row"
    >
      <Button
        handleEvent={[Function]}
        value="4"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          4
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="5"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          5
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="6"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          6
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="-"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          -
        </button>
      </Button>
    </div>
    <div
      className="button-list__row"
    >
      <Button
        handleEvent={[Function]}
        value="1"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          1
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="2"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          2
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="3"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          3
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="+"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          +
        </button>
      </Button>
    </div>
    <div
      className="button-list__row"
    >
      <Button
        handleEvent={[Function]}
        value="0"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          0
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="."
      >
        <button
          className="button"
          onClick={[Function]}
        >
          .
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="="
      >
        <button
          className="button"
          onClick={[Function]}
        >
          =
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="/"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          /
        </button>
      </Button>
    </div>
    <div
      className="button-list__row"
    >
      <Button
        handleEvent={[Function]}
        value="AC"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          AC
        </button>
      </Button>
    </div>
  </div>
</ButtonList>
`;

exports[`<ButtonList /> matches snapshot 1`] = `
<ButtonList
  handleEvent={[MockFunction]}
>
  <div
    className="button-list"
  >
    <div
      className="button-list__row"
    >
      <Button
        handleEvent={[Function]}
        value="7"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          7
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="8"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          8
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="9"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          9
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="*"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          *
        </button>
      </Button>
    </div>
    <div
      className="button-list__row"
    >
      <Button
        handleEvent={[Function]}
        value="4"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          4
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="5"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          5
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="6"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          6
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="-"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          -
        </button>
      </Button>
    </div>
    <div
      className="button-list__row"
    >
      <Button
        handleEvent={[Function]}
        value="1"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          1
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="2"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          2
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="3"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          3
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="+"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          +
        </button>
      </Button>
    </div>
    <div
      className="button-list__row"
    >
      <Button
        handleEvent={[Function]}
        value="0"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          0
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="."
      >
        <button
          className="button"
          onClick={[Function]}
        >
          .
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="="
      >
        <button
          className="button"
          onClick={[Function]}
        >
          =
        </button>
      </Button>
      <Button
        handleEvent={[Function]}
        value="/"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          /
        </button>
      </Button>
    </div>
    <div
      className="button-list__row"
    >
      <Button
        handleEvent={[Function]}
        value="AC"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          AC
        </button>
      </Button>
    </div>
  </div>
</ButtonList>
`;
